### PR TITLE
Add comment explaining why cc does not rebuild on env PATH change

### PR DESCRIPTION
**Pull request by NobodyXu**
For https://github.com/rust-lang/cc-rs/pull/1215#discussion_r1806107565

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -3736,6 +3736,7 @@ impl Build {
         if let Some(val) = self.env_cache.read().unwrap().get(v).cloned() {
             return val;
         }
+        // Prevents spurious rebuilds on Windows #1215
```
Comment (RalfJung) [inline src/lib.rs:3739]:
```suggestion
        // Excluding `PATH` prevents spurious rebuilds on Windows, see
        // <https://github.com/rust-lang/cc-rs/pull/1215> for details.
```

Comment (NobodyXu) [reply src/lib.rs]: Thanks!

**Comment by RalfJung** [comment]
LGTM but I can't approve PRs here.

**Comment by NobodyXu** [comment]
Thanks, cc-rs doesn't require approve because I'm the only active maintainer.

**Comment by NobodyXu** [comment]
Thanks, cc-rs doesn't require approve because I'm the only active maintainer.

**Pull request merged**